fiftyone_pipeline_engines_fiftyone/tests/test_usage.py

```diff
@@ -74,21 +74,28 @@ class UsageTests(unittest.TestCase):
     
     @classmethod
     def setUpClass(cls):
-        cls.receiver = ReceiverThread(app)
-        cls.receiver.start()
+        cls.startReceiver()
 
     def setUp(self):
         app.received = []
 
     @classmethod
     def tearDownClass(cls):
-        print("shutting down")
+        cls.stopReceiver()
+
+    @classmethod
+    def startReceiver(cls):
+        cls.receiver = ReceiverThread(app)
+        cls.receiver.start()
+
+    @classmethod
+    def stopReceiver(cls):
         cls.receiver.shutdown()
 
     def waitForUsageThreads(self):
-        for thread in threading.enumerate():
-            if (thread.name == "ShareUsage POST"):
-                thread.join()
+        # necessary for context switching and
+        # allowing the thread pool in ShareUsage to start sending data and release the GIL
+        sleep(0.1)
```

Comment:
Is the test reliable with this sleep? It seems like waiting for the actual thread would be better?

Reply:
Ok, thanks. That sounds reasonable enough.